lib/utils.pm

```diff
@@ -256,7 +256,9 @@ sub systemctl {
     my $expect_false = $args{expect_false} ? '!' : '';
     my @script_params = ("$expect_false systemctl --no-pager $command", timeout => $args{timeout}, fail_message => $args{fail_message});
     if ($args{ignore_failure}) {
-        script_run(@script_params);
+        # '!' should be added in the command to act like assert_script_run
+        # If $expect_false is set, the double '!!' will act like no inversion
```

Comment:
What happens if `$expect_false` is _not_ set?

Reply:
If `$expect_false` is not set `script_run` is executed with a '!' in front of the command, to act like `assert_script_run` (see line 261).

Reply:
I would re-write the call to `script_run` as:

```
script_run($expect_false . $script_params[0], $args{timeout});
```

Although I'm not sure either is needed as `systemctl` is not doing anything with the return code from `script_run`.

Reply:
Yeah, definitely would remove the `'!'` from the `script_run()` call. IMO, it's not needed, and if kept there, perl won't complain, but bash will. Test:

```
cat file
! cat file
!! cat file
```
In the third example, `!!` is expanded to the previous command line in the shell (on bash), which could potentially be anything.

Reply:
As discussed, I let the code as-is and just document the different behaviour between `script_run` and `assert_script_run`.